src/main/java/com/google/sps/data/Search.java

```diff
@@ -85,8 +88,20 @@ public boolean isSearchedTextMatching(String title, String description) {
     return text == null || text.isEmpty() || isTextMatching(title) || isTextMatching(description);
   }
 
-  public boolean isTextMatching(String string) {
-    return string.toLowerCase().contains(text.toLowerCase());
+  public boolean isTextMatching(String stringToMatch) {
+    List<String> stringList = Arrays.asList(stringToMatch.toLowerCase().split("\\s+"));
+    List<String> textList = Arrays.asList(this.text.toLowerCase().split("\\s+"));
+
+    for (String string : stringList) {
+      int length = string.length();
+      for (String text : textList) {
+        int distance = Math.min(Math.min(text.length(), length) / 2, 3);
```

Comment:
why this constant? maybe move to a constant?

Reply:
So the distance we compare to is smaller for shorter words which helps to avoid really different words coming up as results.